nixos/modules/services/web-apps/nextcloud-imaginary.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, config, ... }:
+let
+  cfg = config.services.nextcloud.imaginary;
+in
+{
+  options.services.nextcloud.imaginary = {
+    enable = lib.mkEnableOption "Imaginary";
+  };
+
+  config = lib.mkIf cfg.enable {
+    services = {
+      imaginary = {
```

Comment:
I don't see why this is its own file.
We also configure other related services such as nginx and postgres in nextcloud.nix and this is trivial to not make a difference.